File: app/src/main/java/com/tjb/dwf/DWFApplication.java
package com.tjb.dwf;

import android.app.Application;
import android.content.Context;

import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class DWFApplication extends Application {
    private static Context context;

    public void onCreate() {
        super.onCreate();
        DWFApplication.context = getApplicationContext();
    }

    public static Context getAppContext() {
        return DWFApplication.context;
    }
}

File: app/src/main/java/com/tjb/dwf/DWFApplication.kt
package com.tjb.dwf

import android.app.Application
import com.tjb.dwf.di.AppComponent
import com.tjb.dwf.di.DaggerAppComponent

open class DWFApplication : Application() {
    val appComponent: AppComponent by lazy {
        initializeComponent()
    }

    open fun initializeComponent(): AppComponent {
        return DaggerAppComponent.factory().create(applicationContext)
    }
}
